adm_program/system/classes/phpini.php

```diff
@@ -0,0 +1,201 @@
+     */
+    public static function getBaseDirs()
+    {
+        return explode(PATH_SEPARATOR, ini_get('open_basedir'));
+    }
+
+    /**
+     * @return int
+     */
+    public static function getMemoryLimit()
+    {
+        return self::getBytesFromSize(ini_get('memory_limit'));
+    }
+
+    /**
+     * @return int
+     */
+    public static function getPostMaxSize()
+    {
+        return self::getBytesFromSize(ini_get('post_max_size'));
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isFileUploadEnabled()
+    {
+        return (bool) ini_get('file_uploads');
+    }
+
+    /**
+     * @return string
+     */
+    public static function getFileUploadTmpDir()
+    {
+        return ini_get('upload_tmp_dir');
+    }
+
+    /**
+     * @return int
+     */
+    public static function getFileUploadMaxFileSize()
+    {
+        return self::getBytesFromSize(ini_get('upload_max_filesize'));
+    }
+
+    /**
+     * @return int
+     */
+    public static function getFileUploadMaxFileCount()
+    {
+        return (int) ini_get('max_file_uploads');
+    }
+
+    /**
+     * @return int
+     */
+    public static function getUploadMaxSize()
+    {
+        return min(self::getMemoryLimit(), self::getPostMaxSize(), self::getFileUploadMaxFileSize());
+    }
+
+    /**
+     * @return bool
+     */
+    public static function checkSizeLimits()
+    {
+        return (self::getMemoryLimit() === -1 || self::getMemoryLimit() >= self::getPostMaxSize())
+            && (self::getPostMaxSize() === -1 || self::getPostMaxSize() >= self::getFileUploadMaxFileSize());
+    }
+
+    /**
+     * @param string $dirPath
+     * @return bool
+     */
+    private static function isInBaseDirs($dirPath)
+    {
+        $baseDirs = self::getBaseDirs();
+
+        if ($baseDirs[0] === '')
+        {
+            return true;
+        }
+
+        $isInBaseDirs = false;
+        foreach ($baseDirs as $baseDir)
+        {
+            if (strpos($dirPath, $baseDir) === 0)
+            {
+                $isInBaseDirs = true;
+            }
+        }
+
+        return $isInBaseDirs;
+    }
+
+    /**
+     * @param string[] $dirPaths
+     * @return bool|string
+     * @throws Exception
+     */
+    public static function setBaseDirs(array $dirPaths = array())
+    {
+        $realDirPaths = array_map('realpath', $dirPaths);
+
+        foreach ($realDirPaths as $realDirPath)
+        {
+            if ($realDirPath === false)
+            {
+                throw new Exception('Not a valid or allowed directory');
+            }
+            if (!self::isInBaseDirs($realDirPath))
+            {
+                throw new Exception('Not in base-directory!');
+            }
+        }
+
+        return ini_set('open_basedir', implode(PATH_SEPARATOR, $realDirPaths));
+    }
+
+    /**
+     * @param string $dirPath
+     * @return bool|string
+     * @throws Exception
+     */
+    public static function setFileUploadTmpDir($dirPath)
```

Comment:
This method is not called yet !?!
Please add an description of the method.

Reply:
I already have plans for this function in the future.